dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/boot/ServerStartJavaEnvironmentVariablesTest.java

```diff
@@ -0,0 +1,138 @@
+    @Test
+    public void testServerStartJavaOptionsSet() throws Exception {
+        Log.entering(c, testName.getMethodName());
+
+        String[] parms = new String[] { "start", SERVER_NAME };
+
+        Properties envVars = new Properties();
+        envVars.put("CDPATH", ".");
+
+        ProgramOutput po = server.getMachine().execute(serverCommand, parms, executionDir, envVars);
+        Log.info(c, testName.getMethodName(), "server start stdout = " + po.getStdout());
+        Log.info(c, testName.getMethodName(), "server start stderr = " + po.getStderr());
+
+        server.waitForStringInLog("CWWKF0011I");
+        server.resetStarted();
+
+        ProgramOutput dumpOut = server.serverDump();
+
+        File[] filesAfterDump = new File(executionDir + "/usr/servers/" + SERVER_NAME).listFiles();
+
+        File dumpFile = new File("");
+        for (File f : filesAfterDump) {
+            String fileName = f.getName();
+            if (fileName.startsWith(SERVER_NAME + ".dump") && fileName.endsWith(".zip")) {
+                dumpFile = f;
+                break;
+            }
+        }
+        assertTrue("The Dump File was not found", dumpFile.getPath().compareTo("") != 0);
+
+        ZipFile zipFile = new ZipFile(dumpFile);
+        for (Enumeration<? extends ZipEntry> en = zipFile.entries(); en.hasMoreElements();) {
+            ZipEntry entry = en.nextElement();
+            if (entry.getName().endsWith("EnvironmentVariables.txt")) {
+                InputStream entryInputStream = zipFile.getInputStream(entry);
+                BufferedReader entryReader = new BufferedReader(new InputStreamReader(entryInputStream));
+
+                String[] firstLine = new String[2], secondLine = new String[2];
+                while ((firstLine = entryReader.readLine().split("=", 2)) != null) {
+                    if (firstLine[0].startsWith("IBM_JAVA_OPTIONS")) {
+                        while ((secondLine = entryReader.readLine().split("=", 2)) != null) {
+                            if (secondLine[0].equals("OPENJ9_JAVA_OPTIONS")) {
+                                Log.info(c, testName.getMethodName(), String.format("%-20s=%s", firstLine[0], firstLine[1]));
+                                Log.info(c, testName.getMethodName(), String.format("%-20s=%s", secondLine[0], secondLine[1]));
+                                assertTrue("IBM_JAVA_OPTIONS did not equal OPENJ9_JAVA OPTIONS", firstLine[1].equals(secondLine[1]));
+                                break;
+                            }
+                        }
+                        assertTrue("OPENJ9_JAVA_OPTIONS was not found", secondLine[0].equals("OPENJ9_JAVA_OPTIONS"));
+                        break;
+                    }
+                }
+                assertTrue("IBM_JAVA_OPTIONS was not found", firstLine[0].equals("IBM_JAVA_OPTIONS"));
+
+                entryReader.close();
+                entryInputStream.close();
+            }
+        }
+
+        zipFile.close();
```

Comment:
You may want to use a try block and have the zipFile.close() and dumofile.delete() in a finally clause so it always gets called.